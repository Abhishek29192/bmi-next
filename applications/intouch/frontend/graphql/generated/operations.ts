import type * as SchemaTypes from "@bmi/intouch-api-types";

export type AddressLinesFragmentFragment = {
  readonly __typename?: "Address";
} & Pick<
  SchemaTypes.Address,
  "firstLine" | "secondLine" | "town" | "region" | "country" | "postcode"
>;

export type CompanyCertificationsFragment = {
  readonly __typename?: "Company";
} & Pick<SchemaTypes.Company, "certifications">;

export type ContactDetailsCollectionFragmentFragment = {
  readonly __typename?: "ContactDetailsCollection";
} & {
  readonly items: ReadonlyArray<
    SchemaTypes.Maybe<
      { readonly __typename?: "ContactDetails" } & Pick<
        SchemaTypes.ContactDetails,
        "fullName" | "subHeading" | "email" | "phoneNumber"
      >
    >
  >;
};

export type UpdateCompanyDetailsMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateCompanyInput;
}>;

export type UpdateCompanyDetailsMutation = {
  readonly __typename?: "Mutation";
} & {
  readonly updateCompany?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateCompanyPayload" } & {
      readonly company?: SchemaTypes.Maybe<
        { readonly __typename?: "Company" } & CompanyDetailsFragmentFragment
      >;
    }
  >;
};

export type GetGlobalDataQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type GetGlobalDataQuery = { readonly __typename?: "Query" } & {
  readonly marketContentCollection?: SchemaTypes.Maybe<
    { readonly __typename?: "MarketContentCollection" } & {
      readonly items: ReadonlyArray<
        SchemaTypes.Maybe<
          { readonly __typename?: "MarketContent" } & Pick<
            SchemaTypes.MarketContent,
            "externalLinkUrl" | "externalLinkLabel"
          > & {
              readonly footerLinksCollection?: SchemaTypes.Maybe<
                {
                  readonly __typename?: "MarketContentFooterLinksCollection";
                } & {
                  readonly items: ReadonlyArray<
                    SchemaTypes.Maybe<
                      { readonly __typename?: "ContentArticle" } & Pick<
                        SchemaTypes.ContentArticle,
                        "title" | "relativePath"
                      >
                    >
                  >;
                }
              >;
              readonly contactUsPage?: SchemaTypes.Maybe<
                { readonly __typename?: "ContentArticle" } & Pick<
                  SchemaTypes.ContentArticle,
                  "title" | "relativePath"
                >
              >;
            }
        >
      >;
    }
  >;
};

export type CompanyAdminsFragmentFragment = {
  readonly __typename?: "Company";
} & {
  readonly companyMembers: {
    readonly __typename?: "CompanyMembersConnection";
  } & {
    readonly nodes: ReadonlyArray<
      { readonly __typename?: "CompanyMember" } & {
        readonly account?: SchemaTypes.Maybe<
          { readonly __typename?: "Account" } & Pick<
            SchemaTypes.Account,
            | "role"
            | "id"
            | "firstName"
            | "lastName"
            | "phone"
            | "email"
            | "photo"
          >
        >;
      }
    >;
  };
};

export type InviteMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.InviteInput;
}>;

export type InviteMutation = { readonly __typename?: "Mutation" } & {
  readonly invite?: SchemaTypes.Maybe<
    ReadonlyArray<
      SchemaTypes.Maybe<
        { readonly __typename?: "Invitation" } & Pick<
          SchemaTypes.Invitation,
          "id" | "invitee"
        > & {
            readonly senderAccount?: SchemaTypes.Maybe<
              { readonly __typename?: "Account" } & Pick<
                SchemaTypes.Account,
                "email"
              >
            >;
          }
      >
    >
  >;
};

export type GetProjectQueryVariables = SchemaTypes.Exact<{
  projectId: SchemaTypes.Scalars["Int"];
}>;

export type GetProjectQuery = { readonly __typename?: "Query" } & {
  readonly project?: SchemaTypes.Maybe<
    { readonly __typename?: "Project" } & Pick<
      SchemaTypes.Project,
      | "id"
      | "name"
      | "technology"
      | "roofArea"
      | "startDate"
      | "endDate"
      | "description"
      | "buildingOwnerFirstname"
      | "buildingOwnerLastname"
      | "buildingOwnerCompany"
      | "buildingOwnerMail"
    > & {
        readonly siteAddress?: SchemaTypes.Maybe<
          { readonly __typename?: "Address" } & Pick<
            SchemaTypes.Address,
            "firstLine" | "secondLine" | "town" | "region" | "postcode"
          >
        >;
        readonly buildingOwnerAddress?: SchemaTypes.Maybe<
          { readonly __typename?: "Address" } & Pick<
            SchemaTypes.Address,
            "firstLine" | "secondLine" | "town" | "region" | "postcode"
          >
        >;
        readonly guarantees: {
          readonly __typename?: "GuaranteesConnection";
        } & {
          readonly nodes: ReadonlyArray<
            { readonly __typename?: "Guarantee" } & Pick<
              SchemaTypes.Guarantee,
              "id" | "guaranteeTypeId"
            > & {
                readonly guaranteeType?: SchemaTypes.Maybe<
                  { readonly __typename?: "ContentfulGuaranteeType" } & Pick<
                    SchemaTypes.ContentfulGuaranteeType,
                    "name"
                  > & {
                      readonly evidenceCategoriesCollection?: SchemaTypes.Maybe<
                        {
                          readonly __typename?: "ContentfulEvidenceCategoryCollection";
                        } & {
                          readonly items?: SchemaTypes.Maybe<
                            ReadonlyArray<
                              SchemaTypes.Maybe<
                                {
                                  readonly __typename?: "ContentfulEvidenceCategory";
                                } & Pick<
                                  SchemaTypes.ContentfulEvidenceCategory,
                                  "name" | "minimumUploads"
                                >
                              >
                            >
                          >;
                        }
                      >;
                    }
                >;
              }
          >;
        };
        readonly evidenceItems: {
          readonly __typename?: "EvidenceItemsConnection";
        } & {
          readonly nodes: ReadonlyArray<
            { readonly __typename?: "EvidenceItem" } & Pick<
              SchemaTypes.EvidenceItem,
              | "id"
              | "name"
              | "guaranteeId"
              | "evidenceCategoryType"
              | "customEvidenceCategoryId"
            > & {
                readonly customEvidenceCategory?: SchemaTypes.Maybe<
                  { readonly __typename?: "ContentfulEvidenceCategory" } & Pick<
                    SchemaTypes.ContentfulEvidenceCategory,
                    "name" | "minimumUploads"
                  >
                >;
              }
          >;
        };
        readonly notes: { readonly __typename?: "NotesConnection" } & {
          readonly nodes: ReadonlyArray<
            { readonly __typename?: "Note" } & Pick<
              SchemaTypes.Note,
              "id" | "body" | "createdAt"
            >
          >;
        };
      }
  >;
};

export type AccountByEmailQueryVariables = SchemaTypes.Exact<{
  email: SchemaTypes.Scalars["String"];
}>;

export type AccountByEmailQuery = { readonly __typename?: "Query" } & {
  readonly accountByEmail?: SchemaTypes.Maybe<
    { readonly __typename?: "Account" } & Pick<
      SchemaTypes.Account,
      | "id"
      | "role"
      | "marketId"
      | "firstName"
      | "lastName"
      | "email"
      | "doceboUserId"
    > & {
        readonly market?: SchemaTypes.Maybe<
          { readonly __typename?: "Market" } & Pick<
            SchemaTypes.Market,
            | "domain"
            | "language"
            | "doceboCompanyAdminBranchId"
            | "doceboInstallersBranchId"
          >
        >;
        readonly companyMembers: {
          readonly __typename?: "CompanyMembersConnection";
        } & {
          readonly nodes: ReadonlyArray<
            { readonly __typename?: "CompanyMember" } & {
              readonly company?: SchemaTypes.Maybe<
                { readonly __typename?: "Company" } & Pick<
                  SchemaTypes.Company,
                  "id" | "status" | "name"
                >
              >;
            }
          >;
        };
      }
  >;
};

export type CreateAccountMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.CreateAccountInput;
}>;

export type CreateAccountMutation = { readonly __typename?: "Mutation" } & {
  readonly createAccount?: SchemaTypes.Maybe<
    { readonly __typename?: "CreateAccountPayload" } & {
      readonly account?: SchemaTypes.Maybe<
        { readonly __typename?: "Account" } & Pick<
          SchemaTypes.Account,
          "id" | "role" | "email" | "firstName" | "lastName" | "marketId"
        > & {
            readonly market?: SchemaTypes.Maybe<
              { readonly __typename?: "Market" } & Pick<
                SchemaTypes.Market,
                | "language"
                | "doceboCompanyAdminBranchId"
                | "doceboInstallersBranchId"
              >
            >;
            readonly companyMembers: {
              readonly __typename?: "CompanyMembersConnection";
            } & {
              readonly nodes: ReadonlyArray<
                { readonly __typename?: "CompanyMember" } & {
                  readonly company?: SchemaTypes.Maybe<
                    { readonly __typename?: "Company" } & Pick<
                      SchemaTypes.Company,
                      "id" | "status"
                    >
                  >;
                }
              >;
            };
          }
      >;
    }
  >;
};

export type CreateDoceboUserMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UserCreateInput;
}>;

export type CreateDoceboUserMutation = { readonly __typename?: "Mutation" } & {
  readonly createDoceboUser?: SchemaTypes.Maybe<
    { readonly __typename?: "UserCreateResponse" } & Pick<
      SchemaTypes.UserCreateResponse,
      "success" | "user_id"
    >
  >;
};

export type UpdateAccountMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateAccountInput;
}>;

export type UpdateAccountMutation = { readonly __typename?: "Mutation" } & {
  readonly updateAccount?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateAccountPayload" } & {
      readonly account?: SchemaTypes.Maybe<
        { readonly __typename?: "Account" } & Pick<
          SchemaTypes.Account,
          "id" | "doceboUserId"
        >
      >;
    }
  >;
};

export type UserByEmailQueryVariables = SchemaTypes.Exact<{
  email: SchemaTypes.Scalars["String"];
}>;

export type UserByEmailQuery = { readonly __typename?: "Query" } & {
  readonly userByEmail?: SchemaTypes.Maybe<
    { readonly __typename?: "UserData" } & Pick<SchemaTypes.UserData, "user_id">
  >;
};

export type InvitationsQueryVariables = SchemaTypes.Exact<{
  invitee: SchemaTypes.Scalars["String"];
}>;

export type InvitationsQuery = { readonly __typename?: "Query" } & {
  readonly invitations?: SchemaTypes.Maybe<
    { readonly __typename?: "InvitationsConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "Invitation" } & Pick<
          SchemaTypes.Invitation,
          "id" | "status" | "invitee" | "senderAccountId"
        >
      >;
    }
  >;
};

export type CompleteInvitationMutationVariables = SchemaTypes.Exact<{
  companyId: SchemaTypes.Scalars["Int"];
}>;

export type CompleteInvitationMutation = {
  readonly __typename?: "Mutation";
} & {
  readonly completeInvitation?: SchemaTypes.Maybe<
    { readonly __typename?: "Account" } & Pick<
      SchemaTypes.Account,
      "id" | "role" | "email" | "firstName" | "lastName" | "marketId"
    > & {
        readonly market?: SchemaTypes.Maybe<
          { readonly __typename?: "Market" } & Pick<
            SchemaTypes.Market,
            | "language"
            | "domain"
            | "doceboCompanyAdminBranchId"
            | "doceboInstallersBranchId"
          >
        >;
      }
  >;
};

export type CreateSsoUrlMutationVariables = SchemaTypes.Exact<{
  username: SchemaTypes.Scalars["String"];
  path?: SchemaTypes.Maybe<SchemaTypes.Scalars["String"]>;
}>;

export type CreateSsoUrlMutation = { readonly __typename?: "Mutation" } & {
  readonly createSSOUrl?: SchemaTypes.Maybe<
    { readonly __typename?: "SSOUrlOutput" } & Pick<
      SchemaTypes.SsoUrlOutput,
      "url"
    >
  >;
};

export type ProductsAndSystemsQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type ProductsAndSystemsQuery = { readonly __typename?: "Query" } & {
  readonly products?: SchemaTypes.Maybe<
    { readonly __typename?: "ProductsConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "Product" } & Pick<
          SchemaTypes.Product,
          | "id"
          | "bmiRef"
          | "family"
          | "name"
          | "brand"
          | "published"
          | "description"
        >
      >;
    }
  >;
  readonly systems?: SchemaTypes.Maybe<
    { readonly __typename?: "SystemsConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "System" } & Pick<
          SchemaTypes.System,
          "id" | "bmiRef" | "description" | "name" | "published"
        >
      >;
    }
  >;
};

export type BulkImportMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.BulkImportInput;
}>;

export type BulkImportMutation = { readonly __typename?: "Mutation" } & {
  readonly bulkImport?: SchemaTypes.Maybe<
    { readonly __typename?: "ImportPayload" } & {
      readonly systemsToInsert?: SchemaTypes.Maybe<
        ReadonlyArray<
          SchemaTypes.Maybe<
            { readonly __typename?: "System" } & Pick<
              SchemaTypes.System,
              "bmiRef"
            >
          >
        >
      >;
      readonly systemsToUpdate?: SchemaTypes.Maybe<
        ReadonlyArray<
          SchemaTypes.Maybe<
            { readonly __typename?: "System" } & Pick<
              SchemaTypes.System,
              "bmiRef"
            >
          >
        >
      >;
      readonly productsToInsert?: SchemaTypes.Maybe<
        ReadonlyArray<
          SchemaTypes.Maybe<
            { readonly __typename?: "Product" } & Pick<
              SchemaTypes.Product,
              "bmiRef"
            >
          >
        >
      >;
      readonly productsToUpdate?: SchemaTypes.Maybe<
        ReadonlyArray<
          SchemaTypes.Maybe<
            { readonly __typename?: "Product" } & Pick<
              SchemaTypes.Product,
              "bmiRef"
            >
          >
        >
      >;
    }
  >;
};

export type UpdateProductMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateProductInput;
}>;

export type UpdateProductMutation = { readonly __typename?: "Mutation" } & {
  readonly updateProduct?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateProductPayload" } & {
      readonly product?: SchemaTypes.Maybe<
        { readonly __typename?: "Product" } & Pick<SchemaTypes.Product, "id">
      >;
    }
  >;
};

export type UpdateSystemMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateSystemInput;
}>;

export type UpdateSystemMutation = { readonly __typename?: "Mutation" } & {
  readonly updateSystem?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateSystemPayload" } & {
      readonly system?: SchemaTypes.Maybe<
        { readonly __typename?: "System" } & Pick<SchemaTypes.System, "id">
      >;
    }
  >;
};

export type CreateCompanyMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateCompanyInput;
}>;

export type CreateCompanyMutation = { readonly __typename?: "Mutation" } & {
  readonly updateCompany?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateCompanyPayload" } & {
      readonly company?: SchemaTypes.Maybe<
        { readonly __typename?: "Company" } & Pick<SchemaTypes.Company, "name">
      >;
    }
  >;
};

export type CurrentCompanyQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type CurrentCompanyQuery = { readonly __typename?: "Query" } & Pick<
  SchemaTypes.Query,
  "currentCompany"
>;

export type GetCurrentCompanyQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type GetCurrentCompanyQuery = { readonly __typename?: "Query" } & Pick<
  SchemaTypes.Query,
  "currentCompany"
>;

export type GetCompanyQueryVariables = SchemaTypes.Exact<{
  companyId: SchemaTypes.Scalars["Int"];
}>;

export type GetCompanyQuery = { readonly __typename?: "Query" } & {
  readonly company?: SchemaTypes.Maybe<
    { readonly __typename?: "Company" } & CompanyDetailsFragmentFragment
  >;
  readonly contactDetailsCollection?: SchemaTypes.Maybe<
    {
      readonly __typename?: "ContactDetailsCollection";
    } & ContactDetailsCollectionFragmentFragment
  >;
};

export type CompanyDetailsFragmentFragment = {
  readonly __typename?: "Company";
} & Pick<
  SchemaTypes.Company,
  | "id"
  | "name"
  | "logo"
  | "phone"
  | "website"
  | "aboutUs"
  | "taxNumber"
  | "tier"
  | "businessType"
  | "ownerFullname"
  | "ownerEmail"
  | "ownerPhone"
  | "publicEmail"
  | "linkedIn"
  | "facebook"
  | "referenceNumber"
> & {
    readonly tradingAddress?: SchemaTypes.Maybe<
      { readonly __typename?: "Address" } & {
        readonly coordinates?: SchemaTypes.Maybe<
          { readonly __typename?: "Point" } & Pick<SchemaTypes.Point, "x" | "y">
        >;
      } & AddressLinesFragmentFragment
    >;
    readonly registeredAddress?: SchemaTypes.Maybe<
      { readonly __typename?: "Address" } & AddressLinesFragmentFragment
    >;
  } & CompanyAdminsFragmentFragment &
  CompanyCertificationsFragment;

export type ImageFragmentFragment = { readonly __typename?: "Asset" } & Pick<
  SchemaTypes.Asset,
  | "title"
  | "description"
  | "contentType"
  | "fileName"
  | "size"
  | "url"
  | "width"
  | "height"
>;

export type GetPartnerBrandsQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type GetPartnerBrandsQuery = { readonly __typename?: "Query" } & {
  readonly marketContentCollection?: SchemaTypes.Maybe<
    { readonly __typename?: "MarketContentCollection" } & {
      readonly items: ReadonlyArray<
        SchemaTypes.Maybe<
          { readonly __typename?: "MarketContent" } & {
            readonly partnerBrandsCollection?: SchemaTypes.Maybe<
              {
                readonly __typename?: "MarketContentPartnerBrandsCollection";
              } & {
                readonly items: ReadonlyArray<
                  SchemaTypes.Maybe<
                    { readonly __typename?: "PartnerBrand" } & Pick<
                      SchemaTypes.PartnerBrand,
                      "name" | "shortDescription"
                    > & {
                        readonly image?: SchemaTypes.Maybe<
                          {
                            readonly __typename?: "Asset";
                          } & ImageFragmentFragment
                        >;
                        readonly logo?: SchemaTypes.Maybe<
                          {
                            readonly __typename?: "Asset";
                          } & ImageFragmentFragment
                        >;
                      }
                  >
                >;
              }
            >;
          }
        >
      >;
    }
  >;
};

export type GetProjectsQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type GetProjectsQuery = { readonly __typename?: "Query" } & {
  readonly projects?: SchemaTypes.Maybe<
    { readonly __typename?: "ProjectsConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "Project" } & Pick<
          SchemaTypes.Project,
          "id" | "name" | "technology" | "startDate" | "endDate"
        > & {
            readonly siteAddress?: SchemaTypes.Maybe<
              { readonly __typename?: "Address" } & Pick<
                SchemaTypes.Address,
                "town" | "postcode"
              >
            >;
          }
      >;
    }
  >;
};

export type CompanyMembersQueryVariables = SchemaTypes.Exact<{
  expiryDate?: SchemaTypes.Maybe<SchemaTypes.Scalars["Datetime"]>;
}>;

export type CompanyMembersQuery = { readonly __typename?: "Query" } & {
  readonly companyMembers?: SchemaTypes.Maybe<
    { readonly __typename?: "CompanyMembersConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "CompanyMember" } & Pick<
          SchemaTypes.CompanyMember,
          "id"
        > & {
            readonly company?: SchemaTypes.Maybe<
              { readonly __typename?: "Company" } & Pick<
                SchemaTypes.Company,
                "name"
              >
            >;
            readonly account?: SchemaTypes.Maybe<
              { readonly __typename?: "Account" } & Pick<
                SchemaTypes.Account,
                | "id"
                | "role"
                | "email"
                | "phone"
                | "photo"
                | "lastName"
                | "firstName"
              > & {
                  readonly certificationsByDoceboUserId: {
                    readonly __typename?: "CertificationsConnection";
                  } & {
                    readonly nodes: ReadonlyArray<
                      { readonly __typename?: "Certification" } & Pick<
                        SchemaTypes.Certification,
                        "id" | "name" | "technology" | "expiryDate"
                      >
                    >;
                  };
                }
            >;
          }
      >;
    }
  >;
};

export type TrainingQueryVariables = SchemaTypes.Exact<{
  catalogueId?: SchemaTypes.Maybe<SchemaTypes.Scalars["Int"]>;
  userId?: SchemaTypes.Maybe<SchemaTypes.Scalars["Int"]>;
}>;

export type TrainingQuery = { readonly __typename?: "Query" } & {
  readonly trainingContentCollection?: SchemaTypes.Maybe<
    { readonly __typename?: "TrainingContentCollection" } & {
      readonly items: ReadonlyArray<
        SchemaTypes.Maybe<
          { readonly __typename?: "TrainingContent" } & Pick<
            SchemaTypes.TrainingContent,
            | "pageHeading"
            | "description"
            | "lmsCtaLabel"
            | "pageSubHeading"
            | "step1Heading"
            | "step1SubHeading"
            | "step1Description"
            | "step2Heading"
            | "step2SubHeading"
            | "step2Description"
            | "step3Heading"
            | "step3SubHeading"
            | "step3Description"
            | "live"
          > & {
              readonly image?: SchemaTypes.Maybe<
                { readonly __typename?: "Asset" } & Pick<
                  SchemaTypes.Asset,
                  "url"
                >
              >;
            }
        >
      >;
    }
  >;
  readonly courseCatalogues?: SchemaTypes.Maybe<
    { readonly __typename?: "CourseCataloguesConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "CourseCatalogue" } & {
          readonly course?: SchemaTypes.Maybe<
            { readonly __typename?: "Course" } & Pick<
              SchemaTypes.Course,
              | "courseId"
              | "name"
              | "technology"
              | "image"
              | "promoted"
              | "trainingType"
              | "description"
            > & {
                readonly courseEnrollments: {
                  readonly __typename?: "CourseEnrollmentsConnection";
                } & {
                  readonly nodes: ReadonlyArray<
                    { readonly __typename?: "CourseEnrollment" } & Pick<
                      SchemaTypes.CourseEnrollment,
                      "id" | "status" | "url" | "courseId"
                    >
                  >;
                };
              }
          >;
        }
      >;
    }
  >;
};

export type DoceboCatalogIdByMarketDomainQueryVariables = SchemaTypes.Exact<{
  domain: SchemaTypes.Scalars["String"];
}>;

export type DoceboCatalogIdByMarketDomainQuery = {
  readonly __typename?: "Query";
} & {
  readonly marketByDomain?: SchemaTypes.Maybe<
    { readonly __typename?: "Market" } & Pick<
      SchemaTypes.Market,
      "doceboCatalogueId"
    >
  >;
};
