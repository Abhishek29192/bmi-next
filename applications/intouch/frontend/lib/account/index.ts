import axios from "axios";
import { v4 } from "uuid";

export const mutationCreateAccount = `mutation CreateAccount($input: CreateAccountInput!) {
  createAccount(input: $input) {
    account {
      id
      marketId
    }
  }
}`;
export const createAccount = async (req, session) => {
  const { AUTH0_NAMESPACE } = process.env;
  const { user } = session;

  // const market = user[`${AUTH0_NAMESPACE}/market`];
  const firstName = user[`${AUTH0_NAMESPACE}/firstname`];
  const lastName = user[`${AUTH0_NAMESPACE}/lastname`];
  const type = user[`${AUTH0_NAMESPACE}/type`];
  const market = user[`${AUTH0_NAMESPACE}/market`];

  const body = {
    query: mutationCreateAccount,
    variables: {
      input: {
        firstName,
        lastName,
        email: user.email,
        role: type === "company" ? "COMPANY_ADMIN" : "INSTALLER",
        marketCode: market
      }
    }
  };

  try {
    const { data } = await axios.post(
      `http://${req.headers.host}/api/graphql`,
      body,
      {
        headers: {
          authorization: `Bearer ${session.accessToken}`,
          "x-request-id": v4()
        }
      }
    );

    return data;
  } catch (error) {
    // eslint-disable-next-line no-console
    console.log("Error:message: ", error.message);
  }
};
