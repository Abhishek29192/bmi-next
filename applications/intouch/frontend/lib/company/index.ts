import { gql } from "@apollo/client";

export const CONTACT_DETAILS_COLLECTION_FRAGMENT = gql`
  fragment ContactDetailsCollectionFragment on ContactDetailsCollection {
    items {
      fullName
      subHeading
      email
      phoneNumber
    }
  }
`;

export const COMPANY_DETAILS_FRAGMENT = gql`
  fragment CompanyDetailsFragment on Company {
    name
    businessType
    logo
    aboutUs
    tradingAddress {
      id
      ...AddressLinesFragment
      # These are required for the Alert banner
      coordinates {
        x
        y
      }
    }
    ownerFullname
    ownerPhone
    ownerEmail
    phone
    publicEmail
    website
    facebook
    linkedIn
  }
`;

export const CompanyRegisteredDetailsFragment = gql`
  fragment CompanyRegisteredDetailsFragment on Company {
    name
    referenceNumber
    registeredAddress {
      id
      ...AddressLinesFragment
    }
    taxNumber
    tier
    companyOperationsByCompany {
      nodes {
        id
        operation
      }
    }
  }
`;

export const COMPANY_PAGE_DETAILS_FRAGMENT = gql`
  fragment CompanyPageDetailsFragment on Company {
    id
    ...CompanyDetailsFragment
    ...CompanyRegisteredDetailsFragment
    ...CompanyAdminsFragment
    ...CompanyCertifications
    ...CompanyDocumentsFragment
    status
    isProfileComplete
  }
`;

export const GET_COMPANIES_BY_MARKET = gql`
  query GetCompaniesByMarket($marketId: Int!, $tag: String!) {
    companies(condition: { marketId: $marketId }) {
      nodes {
        ...CompanyPageDetailsFragment
        updatedAt
      }
    }
    contactDetailsCollection(
      where: {
        contentfulMetadata: {
          tags_exists: true
          tags: { id_contains_some: [$tag] }
        }
      }
    ) {
      ...ContactDetailsCollectionFragment
    }
  }
`;

export const GET_COMPANY_PAGE = gql`
  query GetCompany($companyId: Int!, $tag: String!) {
    company(id: $companyId) {
      ...CompanyPageDetailsFragment
    }
    contactDetailsCollection(
      where: {
        contentfulMetadata: {
          tags_exists: true
          tags: { id_contains_some: [$tag] }
        }
      }
    ) {
      ...ContactDetailsCollectionFragment
    }
  }
`;

export const GET_OPERATION_TYPE_COLLECTION = gql`
  query GetOperationTypeCollection($tag: String!) {
    operationTypeCollection(
      where: {
        contentfulMetadata: {
          tags_exists: true
          tags: { id_contains_some: [$tag] }
        }
      }
    ) {
      items {
        type
        displayName
      }
    }
  }
`;
