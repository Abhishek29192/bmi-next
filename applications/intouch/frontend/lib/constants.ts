import { BusinessType, Operation, Role } from "@bmi/intouch-api-types";

export const ROLES: { [key: string]: Role } = {
  COMPANY_ADMIN: "COMPANY_ADMIN",
  INSTALLER: "INSTALLER",
  MARKET_ADMIN: "MARKET_ADMIN",
  SUPER_ADMIN: "SUPER_ADMIN"
};

export const BUSINESS_TYPES: { [key: string]: BusinessType } = {
  CONTRACTOR: "CONTRACTOR",
  ARCHITECT: "ARCHITECT",
  MERCHANT: "MERCHANT",
  CORP_DEVELOPER: "CORP_DEVELOPER"
};

export const OPERATION_TYPES: { [key: string]: Operation } = {
  FLAT: "FLAT",
  PITCHED: "PITCHED",
  SOLAR: "SOLAR",
  BITUMEN: "BITUMEN",
  TILE: "TILE",
  COATER: "COATER",
  GREEN: "GREEN"
};
