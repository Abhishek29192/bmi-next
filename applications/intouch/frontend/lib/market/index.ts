import { gql } from "@apollo/client";

export const queryMarketsByDomain = gql`
  query getMarketsByDomain($domain: String!) {
    markets(condition: { domain: $domain }) {
      nodes {
        id
        name
        cmsSpaceId
        language
        domain
        doceboInstallersBranchId
        doceboCompanyAdminBranchId
        merchandisingUrl
        merchandiseSso
        projectsEnabled
        gtag
        gtagMarketMedia
        optanonClass
        sendName
        sendMailbox
        locationBiasRadiusKm
        geoMiddle {
          x
          y
        }
      }
    }
  }
`;
