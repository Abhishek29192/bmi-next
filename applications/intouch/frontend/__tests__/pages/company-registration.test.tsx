import React from "react";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import { getServerPageGetCompany } from "../../graphql/generated/page";
import {
  createMockRouter,
  renderWithAllProviders
} from "../../lib/tests/utils";
import { GetGlobalDataQuery } from "../../graphql/generated/operations";
import { generateGlobalPageData } from "../../lib/tests/factories/globalPageData";
import { generateAccount } from "../../lib/tests/factories/account";
import { ROLES } from "../../lib/constants";
import CompanyRegistrationPage, {
  getServerSideProps
} from "../../pages/company-registration";
import { generateCompany } from "../../lib/tests/factories/company";
import { generateMarketContext } from "../../lib/tests/factories/market";

jest.mock("../../lib/middleware/withPage", () => ({
  withPage: (fn) => {
    return (ctx) => {
      return fn(ctx);
    };
  }
}));
jest.mock("next-i18next/serverSideTranslations", () => ({
  serverSideTranslations: () => Promise.resolve({})
}));
jest.mock("../../graphql/generated/page", () => ({
  getServerPageGetCompany: jest.fn()
}));

let editCompanyDialogProps;
jest.mock("../../components/Pages/Company/EditCompany/Dialog.tsx", () => ({
  __esModule: true,
  ...jest.requireActual(
    "../../components/Pages/Company/EditCompany/Dialog.tsx"
  ),
  EditCompanyDialog: jest.fn().mockImplementation((props) => {
    editCompanyDialogProps = props;
    return <div>dialog</div>;
  })
}));

describe("CompanyRegistrationPage", () => {
  const globalPageData: GetGlobalDataQuery = generateGlobalPageData();
  const company = generateCompany();
  const market = generateMarketContext();

  let context;
  const mockApolloQuery = jest.fn();

  const account = generateAccount({
    role: ROLES.SUPER_ADMIN,
    hasCompany: true,
    companyTier: "T2"
  });

  beforeEach(() => {
    context = {
      apolloClient: {
        query: mockApolloQuery
      },
      account: account,
      res: {},
      market: {
        id: 1
      },
      req: {
        headers: {
          host: "en.local.intouch:3000"
        }
      }
    };
  });

  it("should redirect to existing company", async () => {
    (getServerPageGetCompany as jest.Mock).mockImplementationOnce(() =>
      Promise.resolve({
        props: {
          account: account,
          data: {
            company
          }
        }
      })
    );

    const result = await getServerSideProps(context);

    expect(result).toEqual({
      redirect: {
        destination: "/companies/1",
        permanent: false
      }
    });
  });

  it("should return company. Default case", async () => {
    const company = generateCompany({ status: "NEW" });
    (getServerPageGetCompany as jest.Mock).mockImplementationOnce(() =>
      Promise.resolve({
        props: {
          account: account,
          data: {
            company
          }
        }
      })
    );

    const result = await getServerSideProps(context);

    expect(result).toEqual({
      props: {
        company
      }
    });
  });

  it("should not return company. Missing company members", async () => {
    context.account.companyMembers = false;

    (getServerPageGetCompany as jest.Mock).mockImplementationOnce(() =>
      Promise.resolve({
        props: {
          account: account,
          data: {
            company
          }
        }
      })
    );

    const result = await getServerSideProps(context);

    expect(result).toEqual({
      redirect: {
        destination: "/",
        permanent: false
      }
    });
  });

  it("render correctly", async () => {
    const { container } = renderWithAllProviders(
      <RouterContext.Provider value={createMockRouter({})}>
        <CompanyRegistrationPage
          globalPageData={globalPageData}
          company={company}
          mapsApiKey={""}
          account={account}
          market={market}
        />
      </RouterContext.Provider>
    );
    editCompanyDialogProps.onCompanyUpdateSuccess();
    expect(container).toMatchSnapshot();
  });
});
