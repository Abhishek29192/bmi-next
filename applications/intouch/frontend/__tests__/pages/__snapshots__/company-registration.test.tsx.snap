// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyRegistrationPage render correctly 1`] = `
<div>
  <div>
    dialog
  </div>
</div>
`;
